### PR TITLE
ENH: Add support for Thorlabs WFS-40 detector

### PR DESCRIPTION
**Pull request by slactjohnson**
## Description
Adds in a new device for the Thorlabs WFS40 detector. 

## Motivation and Context

This device is used in the laser control system. 
## How Has This Been Tested?

Tested using a WFS-40 in the B40 Heinz lab. The component PVs work, but Typhos hangs when attempting to create a screen. This appears to be related to the parent class PCDSAreaDetectorTyphos, rather than the WFS class, but that can be fixed in a different PR. 
## Where Has This Been Documented?

It hasn't, outside of the source. However, I think the component names are pretty self-explanatory.

### CODE REVIEW

File: pcdsdevices/lasers/thorlabsWFS.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    pupil_diameter_y = Cpt(EpicsSignal, 'PupilDiameterY', kind='config')
+
+    pupil_centroid_x_rbv = Cpt(EpicsSignal, 'PupilCentroidX_RBV',
+                               kind='normal')
+    pupil_centroid_y_rbv = Cpt(EpicsSignal, 'PupilCentroidY_RBV',
+                               kind='normal')
+    pupil_diameter_x_rbb = Cpt(EpicsSignal, 'PupilDiameterX_RBV',
+                               kind='normal')
+    pupil_diameter_y_rbv = Cpt(EpicsSignal, 'PupilDiameterY_RBV',
+                               kind='normal')
```
Comment (ZLLentz) [inline pcdsdevices/lasers/thorlabsWFS.py:28]: Are these RBV PVs are meant to be `EpicsSignalRO`?

File: pcdsdevices/lasers/thorlabsWFS.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    pupil_centroid_x = Cpt(EpicsSignal, 'PupilCentroidX', kind='config')
+    pupil_centroid_y = Cpt(EpicsSignal, 'PupilCentroidY', kind='config')
+    pupil_diameter_x = Cpt(EpicsSignal, 'PupilDiameterX', kind='config')
+    pupil_diameter_y = Cpt(EpicsSignal, 'PupilDiameterY', kind='config')
+
+    pupil_centroid_x_rbv = Cpt(EpicsSignal, 'PupilCentroidX_RBV',
+                               kind='normal')
+    pupil_centroid_y_rbv = Cpt(EpicsSignal, 'PupilCentroidY_RBV',
+                               kind='normal')
+    pupil_diameter_x_rbb = Cpt(EpicsSignal, 'PupilDiameterX_RBV',
```
Comment (ZLLentz) [inline pcdsdevices/lasers/thorlabsWFS.py:25]: typo rbb

Comment (slactjohnson) [reply pcdsdevices/lasers/thorlabsWFS.py:25]: D'oh!

File: pcdsdevices/lasers/thorlabsWFS.py (hunk 1 above)
Comment (slactjohnson) [reply pcdsdevices/lasers/thorlabsWFS.py:28]: No, that would be giving me too much credit. I wrote most of this prior to learning about that signal type, and I forgot to update it here.

Comment (slactjohnson) [reply pcdsdevices/lasers/thorlabsWFS.py:28]: Yes, they should be RO.

**Pull request merged**